sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_do_while.py

```diff
@@ -54,4 +54,23 @@ def pipeline_with_do_while_node():
         _from_component_func=True,
     )
     do_while_node.set_limits(max_iteration_count=max_iteration_count)
+
+    def _infer_and_update_body_input_from_mapping():
+        # pylint: disable=protected-access
+        for output_name, body_input in mapping.items():
+            # if input type is specified, no need to infer and skip
+            if body_input.type is not None:
+                continue
+            inferred_type = body.outputs[output_name].type
```

Comment:
Does body outputs always have type here?

Reply:
I understand it should have; however, if it does not, we do not change anything at all.

Reply:
Add one check for output type, will skip following operations to avoid unexpected breaking.